internal/services/network/application_gateway_resource.go

```diff
@@ -3443,6 +3967,50 @@ func expandApplicationGatewayRequestRoutingRules(d *pluginsdk.ResourceData, gate
 	return &results, nil
 }
 
+func expandApplicationGatewayRoutingRules(d *pluginsdk.ResourceData, gatewayID string) (*[]network.ApplicationGatewayRoutingRule, error) {
+	vs := d.Get("routing_rule").(*pluginsdk.Set).List()
+	results := make([]network.ApplicationGatewayRoutingRule, 0)
+
+	for _, raw := range vs {
+		v := raw.(map[string]interface{})
+
+		name := v["name"].(string)
+		ruleType := v["rule_type"].(string)
+		listenerName := v["listener_name"].(string)
+		listenerID := fmt.Sprintf("%s/listeners/%s", gatewayID, listenerName)
+		backendAddressPoolName := v["backend_address_pool_name"].(string)
+		backendSettingsName := v["backend_settings_name"].(string)
+
+		rule := network.ApplicationGatewayRoutingRule{
+			Name: utils.String(name),
+			ApplicationGatewayRoutingRulePropertiesFormat: &network.ApplicationGatewayRoutingRulePropertiesFormat{
+				RuleType: network.ApplicationGatewayRequestRoutingRuleType(ruleType),
+				Listener: &network.SubResource{
+					ID: utils.String(listenerID),
+				},
+			},
+		}
+
+		if backendAddressPoolName != "" {
+			backendAddressPoolID := fmt.Sprintf("%s/backendAddressPools/%s", gatewayID, backendAddressPoolName)
```

Comment:
Can we construct the ID from its builder function provided in the `parse` package?